### PR TITLE
feat: fix debtRepayment issue in Strategy.sol, tests for CMO migration

### PR DESCRIPTION
**Pull request by mil0xeth**
- fix debtRepayment issue in Strategy.sol 
- test to prove viability of strategy with insufficient deltaCredit (test_delta_credit.py)
- tests for CMO migration (test_0cmo.py)

### CODE REVIEW

**Comment by mil0xeth** [comment]
strategychanges
changes to the strategy

File: tests/Mainnet/test_0cmo.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    delta_credit_before_usdt = liquidityPool_usdt.deltaCredit()
+
+    usdc_vault = Contract(old_stargate_usdc.vault())
+    usdt_vault = Contract(old_stargate_usdt.vault())
+    safe = Contract(usdc_vault.governance())
+
+    new_stargate_usdc = strategist.deploy(Strategy, usdc_vault, lp_staker, 0, False, True, "StargateV2-USDC")
+    new_stargate_usdc.setRewards("0x93A62dA5a14C80f265DAbC077fCEE437B1a0Efde", {"from": strategist})
+    new_stargate_usdc.setStrategist(sms, {"from": strategist})
+    new_stargate_usdt = strategist.deploy(Strategy, usdt_vault, lp_staker, 1, False, True, "StargateV2-USDC")
```
Comment (newmickymousse) [inline tests/Mainnet/test_0cmo.py:45]: name should be "StargateV2-USDT"

File: tests/Mainnet/test_delta_credit.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    # 4- Ensure that there is no sufficient deltaCredit
+    liquidityPool = Contract(strategy.liquidityPool())
+    router = Contract(liquidityPool.router())
+    router.instantRedeemLocal(liquidityPool.poolId(), liquidityPool.deltaCredit(), strategist, {"from":token_LP_whale})
+
+    # 5- Call another harvest and see if profit is repported correctly
+    assert liquidityPool.deltaCredit() < amount
+    vault.updateStrategyDebtRatio(strategy.address, 0, {"from": gov})
+    chain.sleep(1)
+    tx = strategy.harvest({"from": gov})
```
Comment (newmickymousse) [inline tests/Mainnet/test_delta_credit.py:33]: we should add an assert that there are no losses and there is still a debtOutstanding.. `assert vault.debtOutsanding(strategy) > 0`. This basically confirms that there is still debt in the strategy that we couldn't withdraw yet.

**Review by newmickymousse — CHANGES_REQUESTED** [review]

File: tests/Mainnet/test_0cmo.py (hunk 1 above)
Comment (mil0xeth) [reply tests/Mainnet/test_0cmo.py:45]: done

File: tests/Mainnet/test_delta_credit.py (hunk 2 above)
Comment (mil0xeth) [reply tests/Mainnet/test_delta_credit.py:33]: done

**Review by newmickymousse — APPROVED** [review]

**Pull request merged**